manifests/0000_90_dns-operator_03_prometheusrules.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: dns
+  namespace: openshift-dns-operator
+  labels:
+    role: alert-rules
+spec:
+  groups:
+    - name: openshift-dns.rules
+      rules:
+      - alert: CoreDNSPanicCount
+        expr: coredns_panic_count_total > 0
+        for: 5m
+        labels:
+          severity: warning
+        annotations:
+          message: |
+            "{{ $value }} CoreDNS panics encountered on {{ $labels.instance }}"
+      - alert: CoreDNSHealthCheckSlow
+        expr: histogram_quantile(.95, sum(rate(coredns_health_request_duration_seconds_bucket[5m])) by (le)) > 10
```

Comment:
Is this saying that &ge;95% of health probes take &gt; 10 seconds?  Would you mind breaking down and explaining the expression?

Reply:
Saw a similar expression in the cluster network operator alerts (https://github.com/openshift/cluster-network-operator/blob/e86bdfc82c02bdd8d6ad04d9cdd28470c49ed4fe/bindata/network/openshift-sdn/alert-rules.yaml#L53).
This `histogram_quantile` function is needed for histogram-based metrics. This expression determines the average time taken per health check request for all requests within the 95th percentile and below. Dropping the top 5% helps to remove some noise, but I suppose is not necessary. We could use 100th or 99th percentile instead if you'd prefer.

Reply:
Is this per-instance?  So if 5 out of 6 instances are healthy and 1 instance is taking &gt;10 seconds for &ge;95% of its health probes, then that raises an alert, right?

Reply:
Ah, I made a mistake. The sum should be `by (instance, le)` not just `by (le)`.

Reply:
But yes, the query is per-instance.